File: Raven.Client.Java/lib/src/main/java/raven/client/linq/IDocumentQueryGenerator.java
package raven.client.linq;

import raven.client.IDocumentQuery;
import raven.client.document.DocumentConvention;

/**
 * Generate a new document query
 */
public interface IDocumentQueryGenerator {
  /**
   * Gets the conventions associated with this query
   * @return
   */
  public DocumentConvention getConventions();

  /**
   * Create a new query for
   */
  public <T> IDocumentQuery<T> luceneQuery(Class<T> clazz, String indexName, boolean isMapReduce);


}

File: src/Raven.Client/Documents/Query/IDocumentQueryGenerator.cs
using Raven.Client.Document;
using Raven.Client.Linq;

namespace Raven.Client.Documents
{
    ///<summary>
    /// Generate a new document query
    ///</summary>
    public interface IDocumentQueryGenerator
    {
        /// <summary>
        /// Gets the conventions associated with this query
        /// </summary>
        DocumentConvention Conventions { get; }

        /// <summary>
        /// Create a new query for <typeparam name="T"/>
        /// </summary>
        IDocumentQuery<T> Query<T>(string indexName, bool isMapReduce);

        /// <summary>
        /// Create a new query for <typeparam name="T"/>
        /// </summary>
        IAsyncDocumentQuery<T> AsyncQuery<T>(string indexName, bool isMapReduce);

        /// <summary>
        /// Generates a query inspector
        /// </summary>
        /// <returns>RavenQueryInspector object</returns>
        RavenQueryInspector<S> CreateRavenQueryInspector<S>();
    }

}
